File: reddio-csharp/reddio.api/V1/Rest/GetRecordResponse.cs
using System;
using System.Collections.Generic;
using Newtonsoft.Json;

namespace Reddio.Api.V1.Rest
{
    public class GetRecordResponse : List<SequenceRecord>
    {
    }

    public class SequenceRecord
    {
        public const int SequenceStatusSubmitted = 0;
        public const int SequenceStatusAccepted = 1;
        public const int SequenceStatusFailed = 2;
        public const int SequenceStatusProved = 3;
        public const int SequenceStatusProvedError = 4;

        public SequenceRecord(string amount, string assetId, string assetName, string assetType, string contractAddress,
            string displayValue, string from, int recordType, long sequenceId, string starkKey, int status, long time,
            string to, string tokenId)
        {
            Amount = amount;
            AssetId = assetId;
            AssetName = assetName;
            AssetType = assetType;
            ContractAddress = contractAddress;
            DisplayValue = displayValue;
            From = from;
            RecordType = recordType;
            SequenceId = sequenceId;
            StarkKey = starkKey;
            Status = status;
            Time = time;
            To = to;
            TokenId = tokenId;
        }

        public SequenceRecord()
        {
        }

        [JsonProperty("amount")] public string Amount { get; set; }
        [JsonProperty("asset_id")] public string AssetId { get; set; }
        [JsonProperty("asset_name")] public string AssetName { get; set; }
        [JsonProperty("asset_type")] public string AssetType { get; set; }
        [JsonProperty("contract_address")] public string ContractAddress { get; set; }
        [JsonProperty("display_value")] public string DisplayValue { get; set; }
        [JsonProperty("from")] public string From { get; set; }
        [JsonProperty("record_type")] public int RecordType { get; set; }
        [JsonProperty("sequence_id")] public Int64 SequenceId { get; set; }
        [JsonProperty("stark_key")] public string StarkKey { get; set; }
        [JsonProperty("status")] public int Status { get; set; }
        [JsonProperty("time")] public Int64 Time { get; set; }
        [JsonProperty("to")] public string To { get; set; }
        [JsonProperty("token_id")] public string TokenId { get; set; }
    }
}

File: reddio-java/reddio-api/src/main/java/com/reddio/api/v1/rest/GetRecordResponse.java
package com.reddio.api.v1.rest;

import com.fasterxml.jackson.annotation.JsonProperty;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.util.ArrayList;


public class GetRecordResponse extends ArrayList<GetRecordResponse.SequenceRecord> {
    @Data
    @NoArgsConstructor
    @AllArgsConstructor(staticName = "of")
    public static class SequenceRecord {

        public static final int SEQUENCE_STATUS_SUBMITTED = 0;
        public static final int SEQUENCE_STATUS_ACCEPTED = 1;
        public static final int SEQUENCE_STATUS_FAILED = 2;
        public static final int SEQUENCE_STATUS_PROVED = 3;
        public static final int SEQUENCE_STATUS_PROVED_ERROR = 4;

        @JsonProperty("amount")
        public String amount;
        @JsonProperty("asset_id")
        public String assetId;
        @JsonProperty("asset_name")
        public String assetName;
        @JsonProperty("asset_type")
        public String assetType;
        @JsonProperty("contract_address")
        public String contractAddress;
        @JsonProperty("display_value")
        public String displayValue;
        @JsonProperty("from")
        public String from;
        @JsonProperty("record_type")
        public String recordType;
        @JsonProperty("sequence_id")
        public long sequenceId;
        @JsonProperty("stark_key")
        public String starkKey;
        @JsonProperty("status")
        public int status;
        @JsonProperty("resp")
        public String resp;
        @JsonProperty("time")
        public long time;
        @JsonProperty("to")
        public String to;
        @JsonProperty("token_id")
        public String tokenId;
    }
}
